packages/playwright-core/src/server/browserContext.ts

```diff
@@ -180,10 +180,13 @@ export abstract class BrowserContext extends SdkObject {
       page = undefined;
     }
 
-    // Unless I do this early, setting extra http headers below does not respond.
+    // Unless dialogs are dismissed, setting extra http headers below does not respond.
+    page?._frameManager.setCloseAllOpeningDialogs(true);
```

Comment:
This call might as well close all currently open dialogs.

Reply:
Separate semantics -> separate call?